index.js

```diff
@@ -1,59 +1,103 @@
+        const ecosystem = config.ecosystem || {};
 
         if (!executorConfig || !(Array.isArray(executorConfig)) || executorConfig.length === 0) {
             throw new Error('No executor config passed in.');
         }
 
         super();
 
-        executorConfig.forEach((plugin) => {
-            let ExecutorPlugin;
+        this.ecosystem = ecosystem;
+        this.defaultPlugin = config.defaultPlugin;
 
-            try {
-                // eslint-disable-next-line global-require, import/no-dynamic-require
-                ExecutorPlugin = require(`screwdriver-executor-${plugin.name}`);
-            } catch (err) {
-                console.error(err);
+        this.executorConfigs = {};
+        executorConfig.forEach((plugin) => {
+            this.executorConfigs[plugin.name] = Object.assign({ ecosystem }, plugin.options);
 
-                return;
+            if (!this.defaultPlugin) {
+                this.defaultPlugin = plugin.name;
             }
+        });
 
-            const options = Object.assign({ ecosystem }, plugin.options); // Add ecosystem to executor options
-            const executorPlugin = new ExecutorPlugin(options);
+        // eslint-disable-next-line new-cap
+        this.queue = new Resque.queue({ connection: config.redisConnection });
 
-            if (defaultPlugin && defaultPlugin === plugin.name) {
-                // set the default plugin to the desired one
-                this.defaultExecutor = executorPlugin;
-            } else if (!defaultPlugin && !this.defaultExecutor) {
-                // When given no default plugin, use the first module we instantiated
-                this.defaultExecutor = executorPlugin;
+        this.connected = false;
+        this.queue.connect(() => {
+            this.connected = true;
+        });
+    }
+
+    /**
+     * Get the full config for a job
+     * @method getFullConfig
+     * @param {Object} buildConfig  build configuration
+     * @return {Object}
+     */
+    getFullConfig(buildConfig) {
+        const annotations = buildConfig.annotations || {};
+        const executorType = getPluginName(
+            annotations[ANNOTATION_EXECUTOR_TYPE] || this.defaultPlugin);
+        const options = {
+            buildConfig,
+            executor: {}
+        };
+
+        options.executor.name = executorType;
+        options.executor.options = this.executorConfigs[executorType];
+
+        Joi.validate(options, SCHEMA_START);
+
+        return options;
+    }
+
+    /**
+     * Verify that the redis connection is active
+     * @method veryifyConnection
+     * @return {Promise}
+     */
+    verifyConnection() {
+        return new Promise((resolve) => {
+            if (this.connected) {
+                return resolve();
             }
 
-            this[plugin.name] = executorPlugin;
+            return this.queue.connect(() => {
+                this.connected = true;
+                resolve();
```

Comment:
This Promise seems to always be resolved. Is there no case where it would need to be rejected like in the case of the connection failing?